File: backend/src/main/java/top/vuhe/portal/service/impl/StatusServiceImpl.java
package top.vuhe.portal.service.impl;

import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl;
import org.springframework.stereotype.Service;
import top.vuhe.entity.equipment.dao.StatusDAO;
import top.vuhe.mapper.StatusMapper;
import top.vuhe.portal.service.intf.StatusService;

import java.util.Date;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * @author zhuhe
 */
@Service("StatusService")
public class StatusServiceImpl extends ServiceImpl<StatusMapper, StatusDAO> implements StatusService {
    private Map<Integer, StatusDAO> map = new HashMap<>(20);
    private Date date;

    @Override
    public Map<Integer, StatusDAO> getStatusInfo() {
        if (map == null) {
            List<StatusDAO> list = list();
            for (StatusDAO dao : list) {
                map.put(dao.getId(), dao);
            }
        }
        return map;
    }
}

File: backend-kt/src/main/kotlin/top/vuhe/sw/portal/service/impl/StatusServiceImpl.kt
package top.vuhe.sw.portal.service.impl

import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl
import org.springframework.stereotype.Service
import top.vuhe.sw.entity.equipment.dao.StatusDAO
import top.vuhe.sw.mapper.StatusMapper
import top.vuhe.sw.portal.service.StatusService

@Service("StatusService")
class StatusServiceImpl : ServiceImpl<StatusMapper, StatusDAO>(), StatusService {
    private val map: MutableMap<Int?, StatusDAO> = HashMap(20)

    override fun getStatusInfo(): Map<Int?, StatusDAO> {
        if (true) {
            val list = list()
            for (dao in list) {
                map[dao.id] = dao
            }
        }
        return map
    }
}